atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/cleaner/KeyValueServicePuncherStoreTest.java

```diff
@@ -113,9 +115,13 @@ private static long mean(long first, long second) {
         return (first + second) / 2;
     }
 
-    private static PuncherStore initializePuncherStore(Map<Long, Long> timestampMap) {
-        PuncherStore puncherStore = KeyValueServicePuncherStore.create(new InMemoryKeyValueService(false));
+    private void initializePuncherStore(Map<Long, Long> timestampMap) {
+        keyValueService = new InMemoryKeyValueService(false);
+        puncherStore = KeyValueServicePuncherStore.create(keyValueService);
         timestampMap.forEach((key, value) -> puncherStore.put(key, value));
-        return puncherStore;
+    }
+
+    private long getMillisForTimestamp(long timestamp) {
+        return KeyValueServicePuncherStore.getMillisForTimestamp(keyValueService, timestamp);
     }
```

Comment:
Probably would be good to have one test for the case where you have to go across a two week boundary to get the right answer (i.e. your timestamp is in an interval between a two week boundary and the first punch that occurred after said boundary - you should return the last one in the previous period).